### PR TITLE
Merge GEOS/mitgcm

### PR DESCRIPTION
**Pull request by sdrabenh**

### CODE REVIEW

**Review by amolod — CHANGES_REQUESTED** [review]
Nothing in moist should be retained from the branch. all as you have it in the new develop. cicethermo changes and the rest we need.

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/aer_cloud.F90
Comment (amolod) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/aer_cloud.F90:1]: Please leave this file as in develop

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/data_txt
Comment (amolod) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/data_txt:1]: Please leave this file as in develop

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/ras.F90
Comment (amolod) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/ras.F90:1]: Please leave this file as in develop

**Review by amolod — CHANGES_REQUESTED** [review]
Please leave all moist gridcomp files as in develop

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/Raster/makebcs/mkMITAquaRaster.F90 [hunk 1]
```diff
@@ -149,7 +147,7 @@ program MAIN
    GridDir = arg
 
    blanklist = 0
-   OPEN(UNIT=iUnit, file='data.exch2', form='formatted', status='old')
+   OPEN(UNIT=iUnit, name='data.exch2', form='formatted', status='old')
```
[the remark this answers is not in the document]
Comment (mathomp4) [reply GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/Raster/makebcs/mkMITAquaRaster.F90:150]: This is the reason the GNU build in CI is failing.

**Comment by sanAkel** [comment]
Adding `DNA` because [this problem](https://github.com/GEOS-ESM/GEOS_OceanGridComp/pull/23#issuecomment-1470065965)

**Comment by amolod** [comment]
@mathomp4 , @sanAkel, i have checked in an update to mitraster to let it compile. as for the other requests i made above (moist from develop, not from the branch), i am not sure how to check or even do this myself. but the ocean grid comp should be ready to go and the dependency removed.

**Comment by sanAkel** [comment]
> @mathomp4 , @sanAkel, i have checked in an update to mitraster to let it compile. as for the other requests i made above (moist from develop, not from the branch), i am not sure how to check or even do this myself. but the ocean grid comp should be ready to go and the dependency removed.

Okay @amolod 
,

**Comment by amolod** [comment]
@sanAkel santha this is working for @yvikhlya  (he told me on friday), so i think we are all set. all is ready for the merge. would you do this please?

**Comment by sanAkel** [comment]
> @sanAkel santha this is working for @yvikhlya  (he told me on friday), so i think we are all set. all is ready for the merge. would you do this please?

@amolod,
If @yvikhlya says it works for him, we can proceed.

As for any merge in _this_ that's up to @sdrabenh or @wmputman.

**Comment by sdrabenh** [comment]
> 

@sanAkel @amolod @yvikhlya it looks as though these 3 PRs ready to be undrafted:
[GEOSgcm_GridComp #724](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/724) (this one)
[GEOSgcm_App #423](https://github.com/GEOS-ESM/GEOSgcm_App/pull/423)
[GEOS_OceanGridComp #23](https://github.com/GEOS-ESM/GEOS_OceanGridComp/pull/23)
If so, shall I undraft them?

**Comment by yvikhlya** [comment]
> If @yvikhlya says it works for him, we can proceed.

Can't confirm this yet.

**Comment by mathomp4** [comment]
@sdrabenh I'm working on identifying the gcm_setup issue hitting the CI.

**Review by zhaobin74 — CHANGES_REQUESTED** [review]
@atrayano @amolod, after the load balancing was implemented for CICEthermo, all exports of ```CICE4ColumnPhys``` grid comp must  go through the relevant buffers in a bunch of *.h files. I don't see  those files are changed. 

Please see comments @HamidehGMAO put in CICE4ColumnPhys [link](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/bf5b0af2424e976050d679c4bdf69a29c10dfaf8/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp/GEOS_CICE4ColumnPhysGridComp.F90#L2196) 

You could also ```grep FCONDBOTN *.h``` in GEOSsaltwaterGridComp/  to see an example.

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py
[the remark this answers is not in the document]
Comment (weiyuan-jiang) [reply GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py:1]: I think we should use  remap_lake_landice_saltwater.py and remap_catchANDcn.py

**Comment by yvikhlya** [comment]
It builds, but it does not produce correct coupled setup.

**Comment by mathomp4** [comment]
> It builds, but it does not produce correct coupled setup.

You are the expert. 😄 

I can say that all four change (UMD, App, GCM, and Ocean) seem to be zero-diff for AMIP in any order. I can also run MOM6 with all four in any order as well, though all I test is "does it run", not "does it run and make good science"!

**Review by amolod — DISMISSED** [review]

**Comment by gmao-rreichle** [comment]
@sdrabenh, the branch still includes
GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py
Unless this file is removed from the branch, I disagree with merging the branch.

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py
[the remark this answers is not in the document]
Comment (amolod) [reply GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py:1]: @sdrabenh is it ok to remove this file?

**Comment by amolod** [comment]
> @sdrabenh, the branch still includes GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py Unless this file is removed from the branch, I disagree with merging the branch.

@sdrabenh are you ok removing this file?

**Comment by sdrabenh** [comment]
> > @sdrabenh, the branch still includes GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py Unless this file is removed from the branch, I disagree with merging the branch.
> 
> @sdrabenh are you ok removing this file?

Yes. I agree with @gmao-rreichle . This is not needed.

**Comment by amolod** [comment]
> > > @sdrabenh, the branch still includes GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py Unless this file is removed from the branch, I disagree with merging the branch.
> > 
> > 
> > @sdrabenh are you ok removing this file?
> 
> Yes. I agree with @gmao-rreichle . This is not needed.

it was removed.

**Comment by gmao-rreichle** [comment]
> it was removed.

Thanks, @amolod  and @sdrabenh.  It's fine with me, then.

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py
[the remark this answers is not in the document]
Comment (amolod) [reply GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/regrid_sfc_rst.py:1]: it was removed

**Review by gmao-rreichle — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Comment by mathomp4** [comment]
The CI is showing a failure in CICE4 when running coupled debug with GNU:

```
At line 3225 of file /root/project/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp/GEOS_CICE4ColumnPhysGridComp.F90
Fortran runtime error: Array bound mismatch for dimension 1 of array 'deltavol1' (1/820)
```

I'm going to try and duplicate on discover. I'll contact @zhaobin74 and maybe @atrayano to figure out the issue. Hopefully quickly.

**Review by mathomp4 — APPROVED** [review]
Re approve after my GNU fix

**Comment by mathomp4** [comment]
Could someone from @GEOS-ESM/ocean-team re-approve this for @sdrabenh? All previous reviews were dismissed when I implemented my fix for GNU

**Comment by sanAkel** [comment]
> Could someone from @GEOS-ESM/ocean-team re-approve this for @sdrabenh? All previous reviews were dismissed when I implemented my fix for GNU

@yvikhlya, I believe you tested this [PR](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/724#discussion_r1157685527) If so, can you please look into @mathomp4's ⬆️ request. Thanks!

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**